update: 서울 데모데이 작업 내용 병합

YamYamee opened the pull request.

ja7811 committed:
update: 캐스트 생성 프롬프트 업데이트
- 모델 gpt-4o로 변경
- 문장 끝에 DELIMITER 붙이도록 변경

ja7811 committed:
update: 파싱 방법 변경
- @를 기준으로 파싱하도록
- 사용자가 스크립트 입력 시엔 그대로 구두점 파싱
 -> 일단 함수 두 개로 나눠놓음.
 나중에 리펙토링

ja7811 committed:
update: 번역 프롬프트 변경

ja7811 committed:
fix: 파싱 및 sentence 저장 오류 수정
- 스크립트 입력과 키워드 입력 함수 분리 제대로 적용 (임시)
- 파싱 시 \n 거르도록 수정
- Sentence::save에서 i++ 두번 하는 오류 수정..

ja7811 committed:
fix: 파싱 오류 수정
- arraylist를 string[]으로 고쳐서 반환

ja7811 committed:
update: 캐스트 생성 프롬프트 수정
- 이름 추가 (일단 OwncastBot으로)

YamYamee committed:
fix: bookmark 싱크 수정

YamYamee committed:
add: 현재 로그인한 유저 정보 가져오는 api 추가, 키워드 가져오는 프롬프트 개선

YamYamee committed:
update: DB에 카테고리 자동으로 추가 하도록 수정

YamYamee committed:
fix: 직접 입력 부분 수정

ja7811 commented on src/main/java/com/umc/owncast/domain/category/Initializer/MainCategoryInitializer.java:
```diff
@@ -0,0 +1,30 @@
+
+@Component
+@RequiredArgsConstructor
+public class MainCategoryInitializer implements CommandLineRunner {
+
+    private final MainCategoryRepository mainCategoryRepository;
+
+    @Override
+    public void run(String... args) throws Exception {
+        if (mainCategoryRepository.count() == 0) { // 데이터가 없는 경우만 추가
+            mainCategoryRepository.save(new MainCategory(1L, "스포츠"));
+            mainCategoryRepository.save(new MainCategory(2L, "음악"));
+            mainCategoryRepository.save(new MainCategory(3L, "책"));
+            mainCategoryRepository.save(new MainCategory(4L, "미술"));
+            mainCategoryRepository.save(new MainCategory(5L, "드라마/영화"));
+            mainCategoryRepository.save(new MainCategory(6L, "음식"));
+            mainCategoryRepository.save(new MainCategory(7L, "시사/뉴스"));
+            mainCategoryRepository.save(new MainCategory(8L, "직접 입력"));
+        }
+    }
+
+}
```
이걸 해주다니 굿입니다👍 

다 좋은데 `count()`가 0일 때만 삽입하는 것 보단
`findAll()`로 레코드 전부 가져와서 카테고리 이름 존재하는지 각각 비교하면서 넣는 게 더 안전할 거 같긴 해요
(SubcategoryInitializer도 마찬가지)

`id`도 마찬가지로 직접 부여하기보단 `save()`하면서 내부적으로 자동 생성되게 하는 게 더 낫지 않나 싶은데
메인 카테고리는 수가 정해져 있으니까 직접 부여하는 것도 괜찮을 거 같기도 하고??
이건 원하시는 쪽으로 합시다

ja7811 commented:
#87 코드 리팩토링까지 마무리한 후 머지합시다 요건 제가 할게요

ja7811 committed:
add: ChatGptServiceDivider 추가

ja7811 committed:
refact: 스크립트 캐스트 생성 관련 함수 리펙토링
- script로 캐스트 생성 시 ChatGptScriptDivider로 문장 구분 (구분자 = @)
- 데모 직전에 createSpeech_scriptInput 등등 _scriptInput으로 분리했던 함수 전부 삭제
- 기존 ParseService::parseSentences는 @Deprecated 마킹한 후 남겨둠 (parseSentencesByPunctuation으로 개명)

ja7811 commented:
전 끝났으니 Initializer 고친 후 괜찮다 싶으면 머지합시당

yuuddin commented on src/main/java/com/umc/owncast/domain/cast/service/ParsingService.java:
```diff
@@ -11,22 +11,36 @@
+        List<String> result = new ArrayList<>();
+        for(String s : sentences){
+            if(s.isBlank()) continue;
+            result.add(s.strip());
```
이런 경우도 꽤 있었나요?? 아님 그냥 안전하게 하는 용도인가요???

ja7811 replied:
안전하게 하려고 넣었어요 혹시나 싶어서 🧐

yuuddin commented on src/main/java/com/umc/owncast/domain/category/Initializer/MainCategoryInitializer.java:
```diff
@@ -0,0 +1,30 @@
+
+@Component
+@RequiredArgsConstructor
+public class MainCategoryInitializer implements CommandLineRunner {
```
main category enum 처리 할게요 이부분 없애주셔도 돼요!!

yuuddin commented on src/main/java/com/umc/owncast/domain/category/Initializer/SubCategoryInitializer.java:
```diff
@@ -0,0 +1,90 @@
+            subCategoryRepository.save(new SubCategory(36L, "채식", foods, false));
+
+            // 시사/뉴스
+            subCategoryRepository.save(new SubCategory(37L, "정치", news, false));
```
좋긴 한데 
궁금한게 있는데 저희 userMade를 사용하는 부분이 있나요...?

ja7811 replied:
정확히 무슨 의미인가요??

yuuddin replied:
저 뒤에 true false 부분이 userMade column인데 이 데이터를 사용하는 부분이 있는건지 궁금해서요!

YamYamee replied:
사용자가 관심사 설정 변경 할 때, 기존 서브 카테고리가 isUserCreated == true 이면 기존 서브 카테고리 삭제하고 새로 만들도록 하려구요!
이렇게 안하면 subCategory가 계속해서 쌓여서...

yuuddin replied:
아아 넵 알겠습니당 :)

yuuddin replied:
음 근데 사용자 A가 펜싱을 추가했다고 가정하고 사용자 B도 펜싱으로 설정하면
B도 해당 subCategory로 mapping이 될텐데 의도하시는 부분을 적용하려면 user를 mapping 하는게 맞지 않을까요...?

ja7811 replied:
> 음 근데 사용자 A가 펜싱을 추가했다고 가정하고 사용자 B도 펜싱으로 설정하면
B도 해당 subCategory로 mapping이 될텐데 의도하시는 부분을 적용하려면 user를 mapping 하는게 맞지 않을까요...?

저도 이 생각이 들긴 하네요

YamYamee replied:
제가 생각했던거는 '펜싱'이라는 이름은 같고 id는 다른 두 개 카테고리를 만드는거였는데... 이러면 별로일까요??
